sdks/java/core/src/test/java/org/apache/beam/sdk/io/WriteFilesTest.java

```diff
@@ -278,6 +280,36 @@ public void testExpandShardedWrite() throws IOException {
         WriteFiles.to(makeSimpleSink()).withNumShards(20));
   }
 
+  /**
+   * Test that WriteFiles with a configured sharding function distributes elements according to
+   * assigned key. Test use simple sharding which co-locate same elements into same shard.
+   */
+  @Test
+  @Category(NeedsRunner.class)
+  public void testCustomShardingFunctionWrite() throws IOException {
```

Comment:
Could we also add a test for `FlinkAutoBalancedShardKeyShardingFunction`?

Reply:
Yes